pydata_center/data/dataset.py

```diff
@@ -0,0 +1,247 @@
+            method (str): How to handle missing values.
+                          'drop' — remove rows with NaNs;
+                          'mean' — fill NaNs with column means.
+
+        Returns:
+            MetricsDataset: The current instance (for method chaining).
+        """
+        if method == 'drop':
+            self.df = self.df.dropna()
+        else:
+            self.df = self.df.fillna(self.df.mean(numeric_only=True))
+        return self
+
+    def create_sequences(
+        self,
+        window: int = 5,
+        scale: bool = False,
+        alert_padding_sec: int = 30
+    ) -> 'SequenceBuilder':
+        """
+        Create windowed sequences for forecasting and classification.
+
+        Args:
+            window (int): Number of past time steps to include per sample.
+            scale (bool): Whether to apply StandardScaler to features.
+            alert_padding_sec (int): Time window (in seconds) around a
+                                     prediction point to check for alerts.
+
+        Returns:
+            SequenceBuilder: Helper object for exporting ML-ready sequences.
+        """
+        return SequenceBuilder(
+            df=self.df,
+            window=window,
+            scale=scale,
+            alert_padding_sec=alert_padding_sec
+        )
+
+
+class SequenceBuilder:
+    """
+    Constructs windowed sequences from metric data for machine learning.
+    Supports both regression (CPU forecast) and binary classification (alert).
+    """
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        window: int,
+        scale: bool,
+        alert_padding_sec: int
+    ):
+        """
+        Initialize a SequenceBuilder with given parameters.
+
+        Args:
+            df (pd.DataFrame): Input metric data.
+            window (int): Number of time steps in each sample window.
+            scale (bool): Whether to normalize features using StandardScaler.
+            alert_padding_sec (int): Number of seconds around each sample
+                                     to detect alert correlation.
+        """
+        self.df = df
+        self.window = window
+        self.scale = scale
+        self.alert_padding = timedelta(seconds=alert_padding_sec)
+
+    def to_numpy(self) -> Tuple[
+        np.ndarray, np.ndarray, np.ndarray, np.ndarray
+    ]:
+        """
+        Convert the dataset into NumPy arrays for ML training.
+
+        Returns:
+            Tuple containing:
+              - X_reg (np.ndarray): Features for regression
+              - y_reg (np.ndarray): CPU target values
+              - X_clf (np.ndarray): Features for classification
+              - y_clf (np.ndarray): Binary alert labels
+        """
+        X_reg, y_reg, X_clf, y_clf = [], [], [], []
+
+        alerts = TriggeredAlert.objects \
+            .filter(rule__is_active=True) \
+            .values_list('triggered_at', flat=True)
+        alert_times = np.array(list(alerts), dtype='datetime64[ns]')
```

Comment:
Consider handling cases where alert_times is empty - currently, this can lead to misleading labels.